### DISCUSSION TITLE
Adding PARAM_GROUP prevents Crazyflie from booting.

### DISCUSSION THREAD

**Discussion by GoLut** [bitcraze/discussions#702] [Q&A] [answered]
Hi, all.

I have a question regarding the process of adding parameters.

Whenever I add the following lines of code to a custom library I included in the Crazyflie firmware the following happens:
The Crazyflie led M2 is solid blue, M3 blinks rapidly, and M4 and M1 only faintly blink red.
I believe it is stuck trying to boot. However, I do not know how to solve it besides removing my PARAM group.

What have I tried:
- Move the param_group to a different file
- Make the parameter volatile (maybe the compiler optimized it out)
- Add an extra parameter to an already existing group for example the estimator_kalman.c file. (still same result)
None of the above have solved the issue.

For what it's worth, I have also encountered this issue before while adding a log_parameter, Creating a separate log_parameter_group seemed to solve it then. Didn't work this time.

```
PARAM_GROUP_START(command_to_drone)
    PARAM_ADD(PARAM_FLOAT, motion_command, &new_recieved_command)
PARAM_GROUP_STOP(command_to_drone)
```

Where the new_recieved_command variable is defined at the top of the .c file as:
```
static float volatile new_recieved_command = 0;
```

How should I approach this issue?

**Comment by krichardsson** [comment, marked as answer]
Hi, 
I think the problem is that the combined length of the group name and parameter name is too long. My guess is that there is a buffer overflow somewhere.

I tested on my Crazyflie and got the same problem with your code, but if I change the param name to "qwe", it works.

**Reply by GoLut** [reply]
Thankyou, That seems to have solved the issue!
